### PR TITLE
Laboratory work #8, Ekaterina Kondrashova - 20FPL1

### PR DESCRIPTION
**Pull request by eskondrashova**

### CODE REVIEW

File: lab_7_llm/assets/index.html [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html>
```
Comment (SalnikovIgor) [inline lab_7_llm/assets/index.html]: Remove all files for lab 7 from the PR. If you want to save them, move them to a separate branch

File: lab_8_llm/main.py [hunk 2]
```diff
@@ -96,6 +118,8 @@ def __getitem__(self, index: int) -> tuple[str, ...]:
         Returns:
             tuple[str, ...]: The item to be received
         """
+        return str(self._data[ColumnNames.SOURCE.value].iloc[index]),\
```
Comment (SalnikovIgor) [inline lab_8_llm/main.py]: Instead of `\` you can put the row obtained using `.iloc` into a separate variable, and then get column values ​​from it

File: lab_8_llm/main.py [hunk 3]
```diff
@@ -105,13 +129,16 @@ def data(self) -> DataFrame:
...
         """
+        return self._data
 
 
 class LLMPipeline(AbstractLLMPipeline):
     """
     A class that initializes a model, analyzes its properties and infers it.
     """
 
+    _model: torch.nn.Module
```
Comment (SalnikovIgor) [inline lab_8_llm/main.py:140]: Are you sure you need this initialization?

File: lab_8_llm/main.py [hunk 4]
```diff
@@ -171,6 +245,23 @@ def _infer_batch(self, sample_batch: Sequence[tuple[str, ...]]) -> list[str]:
...
+
+        for sequence in sample_batch[0]:
+            inputs = self._tokenizer(sequence,
+                                     max_length=self._max_length,
+                                     padding=True,
+                                     truncation=True,
+                                     return_tensors="pt")
+            outputs = self._model(**inputs)
+
+            predicted = torch.nn.functional.softmax(outputs.logits, dim=1)
```
Comment (SalnikovIgor) [inline lab_8_llm/main.py]: Is it possible not to use here `softmax`?

File: lab_8_llm/main.py [hunk 5]
```diff
@@ -171,6 +245,23 @@ def _infer_batch(self, sample_batch: Sequence[tuple[str, ...]]) -> list[str]:
...
+            predicted = torch.argmax(predicted, dim=1).numpy()
```
Comment (SalnikovIgor) [inline lab_8_llm/main.py]: Why convert to `numpy`?

File: lab_8_llm/start.py [hunk 6]
```diff
@@ -1,16 +1,60 @@
...
+                           batch_size=batch_size,
+                           device="cpu")
+    pipeline.analyze_model()
+    pipeline.infer_sample(next(iter(dataset)))
+
+    # mark 8
+    batch_size = 64
+    pipeline = LLMPipeline(model_name=settings_path.parameters.model,
+                           dataset=dataset,
+                           max_length=120,
```
Comment (SalnikovIgor) [inline lab_8_llm/start.py]: If the `batch_size` is in a separate variable, then why is the `max_length` not?

File: requirements.txt [hunk 7]
```diff
@@ -0,0 +1,10 @@
+datasets==2.16.1
```
Comment (SalnikovIgor) [inline requirements.txt:2]: Add line
`--extra-index-url https://download.pytorch.org/whl/cpu`

**Review by SalnikovIgor — CHANGES_REQUESTED** [review]

**Review by SalnikovIgor — CHANGES_REQUESTED** [review]
Fix past comments

**Review by SalnikovIgor — CHANGES_REQUESTED** [review]
Fix past comments

File: requirements.txt (hunk 7 above)
Comment (eskondrashova) [reply requirements.txt:2]: done

File: lab_8_llm/start.py (hunk 6 above)
Comment (eskondrashova) [reply lab_8_llm/start.py]: corrected

File: lab_8_llm/main.py (hunk 2 above)
Comment (eskondrashova) [reply lab_8_llm/main.py]: corrected

File: lab_8_llm/main.py (hunk 4 above)
Comment (eskondrashova) [reply lab_8_llm/main.py]: implementation has been changed

File: lab_8_llm/main.py (hunk 5 above)
Comment (eskondrashova) [reply lab_8_llm/main.py]: implementation has been changed

File: lab_7_llm/assets/index.html (hunk 1 above)
Comment (eskondrashova) [reply lab_7_llm/assets/index.html]: done

File: lab_8_llm/main.py (hunk 3 above)
Comment (eskondrashova) [reply lab_8_llm/main.py:140]: otherwise this error appears:
Argument 1 to "RawDataPreprocessor" has incompatible type "DataFrame | None"; expected "DataFrame"  [arg-type]

File: lab_7_llm/assets/index.html (hunk 1 above)
Comment (SalnikovIgor) [reply lab_7_llm/assets/index.html]: Done?

**Review by SalnikovIgor — CHANGES_REQUESTED** [review]

**Review by SalnikovIgor — CHANGES_REQUESTED** [review]
Checkout the files from 7 lab to the master

File: lab_7_llm/assets/index.html (hunk 1 above)
Comment (eskondrashova) [reply lab_7_llm/assets/index.html]: now yes
